pkg/designatebackendbind9/service.go

```diff
@@ -1,40 +1,66 @@
 package designatebackendbind9
 
 import (
+	"fmt"
 	designatev1beta1 "github.com/openstack-k8s-operators/designate-operator/api/v1beta1"
 	labels "github.com/openstack-k8s-operators/lib-common/modules/common/labels"
-	service "github.com/openstack-k8s-operators/lib-common/modules/common/service"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func Service(m *designatev1beta1.DesignateBackendbind9) *corev1.Service {
-	labels := labels.GetLabels(m, "designatebackendbind9", map[string]string{
+const (
+	Bind9AppName = "designatebackendbind9"
+)
+
+// CreateBindServices creates a service for each bind9 pod
+func CreateBindServices(m *designatev1beta1.DesignateBackendbind9, replicas int32) []*corev1.Service {
+	services := make([]*corev1.Service, replicas)
+
+	for i := int32(0); i < replicas; i++ {
+		podName := fmt.Sprintf("%s-%d", m.Name, i)
+		services[i] = createSingleBindService(m, podName)
+	}
+
+	return services
+}
+
+func createSingleBindService(m *designatev1beta1.DesignateBackendbind9, podName string) *corev1.Service {
+	labels := labels.GetLabels(m, Bind9AppName, map[string]string{
 		"owner": "designate",
 		"cr":    m.GetName(),
-		"app":   "designatebackendbind9",
+		"app":   Bind9AppName,
 	})
-	servicePorts := make([]corev1.ServicePort, 2)
-	servicePorts[0] = corev1.ServicePort{
-		Name:     "named",
-		Port:     53,
-		Protocol: corev1.ProtocolUDP,
-	}
-	servicePorts[1] = corev1.ServicePort{
-		Name:     "named",
-		Port:     53,
-		Protocol: corev1.ProtocolTCP,
-	}
-	details := &service.GenericServiceDetails{
-		Name:      m.GetName(),
-		Namespace: m.GetNamespace(),
-		Labels:    labels,
-		Selector: map[string]string{
-			"app": "bind",
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: m.Namespace,
+			Labels:    labels,
+			Annotations: map[string]string{
+				"metallb.universe.tf/address-pool": "internalapi",
```

Comment:
The address pool will need to be configurable.

Reply:
Done, I will upload another version.